### ISSUE TITLE
example code verification fails

### ISSUE THREAD

**Issue by MeY0u** [codeljo/AA_MCP2515#3]
hello, 
i am have opened the example code for send and receive , ran the varification option in arduino ide and i got many errors:
```

C:\Users\Me\Documents\Arduino\libraries\AA_MCP2515\src\CANIO.cpp: In member function 'void CANIO::begin()':
C:\Users\Me\Documents\Arduino\libraries\AA_MCP2515\src\CANIO.cpp:11:14: error: passing 'const SPIClass' as 'this' argument discards qualifiers [-fpermissive]
   _spi.begin();
 .
.
.
c:\users\me\appdata\local\arduino15\packages\esp32\tools\xtensa-esp32s3-elf-gcc\esp-2021r2-patch5-8.4.0\xtensa-esp32s3-elf\include\c++\8.4.0\bits\stl_algobase.h:195:5: note:   template argument deduction/substitution failed:
C:\Users\Me\Documents\Arduino\libraries\AA_MCP2515\src\CANFrame.cpp:83:51: note:   deduced conflicting types for parameter 'const _Tp' ('unsigned char' and 'int')
   for (uint8_t i=0; i<min(data_length, CAN_DLC_MAX); i++) {
                                                   ^

exit status 1

Compilation error: exit status 1
```

i didnt do anything but open the code. are you familiar with this error?

**Comment by codeljo** [comment]
Hi, I will see if I can reproduce this issue as I believe it is related to ESP32 based boards. Just waiting on a ESP32-S3 so I will let you know asap.

Thank you

**Comment by codeljo** [comment]
I just released v1.0.5 which adds support for ESP32 based boards. The updated library should be available in the Arduino IDE within a few hours. Thanks

**Comment by MeY0u** [comment]
Thank you for the fast update, i will check it later
